File: SwiatZwierzat_Java/Stork.java
public class Stork extends Bird{ //*dziedziczenie* extends uzywamy podczas dziedziczenia klas. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
    //możemy stworzyć oddzielną klasę, czyli Animal i odziedziczyć po niej zestaw cech które ma każde zwierzę

    Stork(){
        super(); //za pomoca slowa super w tym przypadku wywolujemy konstruktor klasy, po ktorej dziedziczymy
        System.out.println("Jestem bocianem");
    }
    void fly(){
        System.out.println("Potrafie latac");
    }

    @Override
    public void presentYourself() {
        System.out.println("Mam na imie " + getName() + " oraz zyje juz od " + getAge() + " lat");
    }
}

File: SwiatZwierzat_C++/Stork.cpp
#pragma once
#include "Bird.cpp"

class Stork : public Bird { //*dziedziczenie* slowo public oznacza, że dziedziczone elementy (np. zmienne lub funkcje) mają taką widoczność jak w klasie bazowej, oprocz private w bazowej. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
				//mozemy stworzyc oddzielna klase, czyli np Bird i odziedziczyc po niej zestaw cech które ma kazdy ptak
public:
    Stork() : Bird() { //wywolujemy konstruktor klasy, po ktorej dziedziczymy, bo bocian tez jest ptakiem
        cout << "Jestem bocianem" << endl;
    }
    void fly(){
        cout << "Potrafie latac" << endl;
    }

    void presentYourself() {
        cout << "Mam na imie " << getName() << " oraz zyje juz od " << getAge() << " lat" << endl;
    }
};
